homework_4/task01/task_1_read_file.py

```diff
@@ -0,0 +1,36 @@
+"""
+Write a function that gets file path as an argument.
+Read the first line of the file.
+If first line is a number return true if number in an interval [1, 3)*
+and false otherwise.
+In case of any error, a ValueError should be thrown.
+
+Write a test for that function using pytest library.
+
+Definition of done:
+ - function is created
+ - function is properly formatted
+ - function has positive and negative tests
+ - all temporary files are removed after test run
+
+You will learn:
+ - how to test Exceptional cases
+ - how to clean up after tests
+ - how to check if file exists**
+ - how to handle*** and raise**** exceptions in test. Use sample from the documentation.
+
+* https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology
+** https://docs.python.org/3/library/os.path.html
+*** https://docs.python.org/3/tutorial/errors.html#handling-exceptions
+**** https://docs.python.org/3/tutorial/errors.html#raising-exceptions
+"""
+
+
+def read_magic_number(path: str) -> bool:
+    try:
+        with open(path, "r") as file_path:
+            first_line = file_path.readline()
+            number_in_line = float(first_line)
+        return 1 <= number_in_line < 3
```

Comment:
this is not a good practice to put everything in try-except block. try to isolate it, make as less code inside as you can

Reply:
becayse `Exception` is a common exception, and it can be thrown from any line of code inside try-except block, it makes debug harder

Reply:
comparison of two numbers will not lead to the exception.

in this case, reading of the file and converting string to float may lead to exception